test/integration/externalnode_test.go

```diff
@@ -0,0 +1,361 @@
+		cloudProvider = strings.Split(cloudProviders, ",")[0]
+		switch cloudProvider {
+		case string(v1alpha1.AWSCloudProvider):
+			account = &v1alpha1.CloudProviderAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testAccountName,
+					Namespace: namespace.Name,
+				},
+				Spec: v1alpha1.CloudProviderAccountSpec{
+					PollIntervalInSeconds: &pollInterval,
+					AWSConfig: &v1alpha1.CloudProviderAccountAWSConfig{
+						Region: accountParameters.Aws.Region,
+						SecretRef: &v1alpha1.SecretReference{
+							Name:      accountParameters.SecretRef.Name,
+							Namespace: accountParameters.SecretRef.Namespace,
+							Key:       accountParameters.SecretRef.Key,
+						},
+					},
+				},
+			}
+		case string(v1alpha1.AzureCloudProvider):
+			account = &v1alpha1.CloudProviderAccount{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      testAccountName,
+					Namespace: namespace.Name,
+				},
+				Spec: v1alpha1.CloudProviderAccountSpec{
+					PollIntervalInSeconds: &pollInterval,
+					AzureConfig: &v1alpha1.CloudProviderAccountAzureConfig{
+						Region: accountParameters.Azure.Location,
+						SecretRef: &v1alpha1.SecretReference{
+							Name:      accountParameters.SecretRef.Name,
+							Namespace: accountParameters.SecretRef.Namespace,
+							Key:       accountParameters.SecretRef.Key,
+						},
+					},
+				},
+			}
+		default:
+			Fail("Invalid Provider Type")
+		}
+		selector = &v1alpha1.CloudEntitySelector{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      testAccountName,
+				Namespace: namespace.Name,
+			},
+			Spec: v1alpha1.CloudEntitySelectorSpec{
+				AccountName: testAccountName,
+				VMSelector:  []v1alpha1.VirtualMachineSelector{},
+			},
+		}
+		// SDK call will automatically base64 encode data.
+		secret = &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      accountParameters.SecretRef.Name,
+				Namespace: accountParameters.SecretRef.Namespace,
+			},
+			Data: map[string][]byte{accountParameters.SecretRef.Key: []byte(accountParameters.SecretRef.Credential)},
+		}
+		createNS()
+		createAccount()
+	})
+
+	AfterEach(func() {
+		result := CurrentGinkgoTestDescription()
+		if result.Failed {
+			if len(supportBundleDir) > 0 {
+				logf.Log.Info("Collect support bundles for test failure.")
+				fileName := utils.GenerateNameFromText(result.FullTestText, testFocus)
+				utils.CollectSupportBundle(kubeCtl, path.Join(supportBundleDir, fileName))
+			}
+			if preserveSetupOnFail {
+				logf.Log.V(1).Info("Preserve setup on failure")
+				return
+			}
+		}
+		preserveSetup = false
+		deleteAccount()
+		deleteNS()
+	})
+
+	table.DescribeTable("Testing agented and agentless configuration through CES",
+		func() {
+			expectedResult := fmt.Sprintf("virtualmachine-%s", testVm)
+
+			By("Configure entity selector with agented field set to true, check ExternalNode creation")
+			tester(expectedResult, false, true, false)
```

Comment:
We should be checking for the ExternalNode deletion, before running this test.
i.e. order should be something like this

1. Configure EntitySelector with Agented=true for VM1, verify En is created.
2. Configure EntitySelector with Agented=false for VM1, verify Ee is created, verify En is deleted.
3. Configure EntitySelector with Agented=true for VM1, verify En is created, verify Ee from step 2 is deleted.
4. Delete EntitySelector and verify En Deletion.

It seems that currently we are missing additional verification in 2nd and 3rd step.

Reply:
Done